### PR TITLE
Add toggle icon

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/panel/issues/6022

To be used for ChatReactionIcons

### CODE REVIEW

**Comment by philippjfr** [comment]
Good start! Next we probably have to create and sync a bunch properties between this wrapper and the underlying icon model.

**Comment by ahuang11** [comment]
Okay this is ready for the next release I think!

ToggleIcon
https://github.com/holoviz/panel/assets/15331990/2c300933-0eba-4b5b-aaa7-6ce15b4a8dfa

ChatMessage
https://github.com/holoviz/panel/assets/15331990/d196967d-6d9d-439f-b331-db822fca28c6

File: panel/chat/icon.py [hunk 1]
```diff
@@ -5,45 +5,15 @@
...
 from ..io.resources import CDN_DIST
-from ..io.state import state
-from ..pane.image import SVG
+from ..layout import Column
 from ..reactive import ReactiveHTML
+from ..widgets.base import CompositeWidget
+from ..widgets.icon import ToggleIcon
 
-# if user cannot connect to internet
-HEART_SVG = """
```
Comment (ahuang11) [inline panel/chat/icon.py:17]: Someday will re-implement cached versions
https://github.com/holoviz/panel/issues/6037

File: panel/models/icon.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    this.model.value = !this.model.value;
+    this.update_icon()
+  }
+
+  connect_signals(): void {
+    super.connect_signals();
+    const { icon, active_icon, value, disabled } = this.model.properties;
+    this.on_change(icon, () => this.update_icon());
+    this.on_change(active_icon, () => this.update_icon());
+    this.on_change(value, () => this.update_icon());
```
Comment (philippjfr) [inline panel/models/icon.ts:47]:
```suggestion
    this.on_change([active_icon, icon, value], () => this.update_icon());
```

File: panel/models/icon.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    super(attrs);
+  }
+
+  static {
+    this.prototype.default_view = ToggleIconView;
+
+    this.define<ToggleIcon.Props>(({ String, Boolean }) => ({
+      icon: [String, "heart"],
+      active_icon: [String, ""],
+      value: [Boolean, false],
```
Comment (philippjfr) [inline panel/models/icon.ts:107]:
```suggestion
      active_icon: [String, ""],
      icon: [String, "heart"],
      value: [Boolean, false],
```

File: panel/widgets/icon.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+class ToggleIcon(Widget):
+
+    icon = param.String(default='heart', doc="""
+        The name of the icon to display from
+        [tabler-icons.io](https://tabler-icons.io)/""")
+
+    active_icon = param.String(default='', doc="""
+        The name of the icon to display when toggled from
+        tabler-icons.io](https://tabler-icons.io)/""")
```
Comment (philippjfr) [inline panel/widgets/icon.py:20]:
```suggestion
    active_icon = param.String(default='', doc="""
        The name of the icon to display when toggled from
        tabler-icons.io](https://tabler-icons.io)/""")

    icon = param.String(default='heart', doc="""
        The name of the icon to display from
        [tabler-icons.io](https://tabler-icons.io)/""")
```

**Review by philippjfr — APPROVED** [review]

**Pull request merged**